Python/pystate.c

```diff
@@ -147,24 +149,24 @@ _PyRuntimeState_ReInitThreads(_PyRuntimeState *runtime)
     PyMemAllocatorEx old_alloc;
     _PyMem_SetDefaultAllocator(PYMEM_DOMAIN_RAW, &old_alloc);
 
-    runtime->interpreters.mutex = PyThread_allocate_lock();
-    runtime->interpreters.main->id_mutex = PyThread_allocate_lock();
-    runtime->xidregistry.mutex = PyThread_allocate_lock();
-
-    PyMem_SetAllocator(PYMEM_DOMAIN_RAW, &old_alloc);
-
-    if (runtime->interpreters.mutex == NULL) {
+    if (_PyThread_at_fork_reinit(&runtime->interpreters.mutex) < 0) {
```

Comment:
I suggest to add 3 local variables to store _PyThread_at_fork_reinit() result and only check the result after `PyMem_SetAllocator(PYMEM_DOMAIN_RAW, &old_alloc);` call. Just to limit code between the two PyMem calls.